version: '3'
services:
  postgres:
    image: postgres:9.4
    volumes:
      - "postgres_data:/var/lib/postgresql/data"
    deploy:
      replicas: 1
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      PGDATA: "/var/lib/postgresql/data/pgdata"
    networks:
      - backend

  redis:
    image: redis:3.2
    deploy:
      replicas: 2
      restart_policy:
        condition: on-failure
        max_attempts: 3
    networks:
      - backend

  django_swarm_app:
    image: django_swarm_app:latest
    build: .
    command: [ "python", "manage.py", "runserver", "0.0.0.0:80"]
    depends_on:
      - postgres
      - redis
    environment:
      - DATABASE_USER=postgres
      - DATABASE_PASSWORD=postgres
      - DATABASE_HOST=postgres
      - DATABASE_PORT=5432
      - REDIS_CACHE_BACKEND=django_redis.cache.RedisCache
      - REDIS_CACHE_LOCATION=redis://redis:6379/
    deploy:
      replicas: 2
      restart_policy:
        condition: on-failure
        max_attempts: 3
      update_config:
        parallelism: 2
        delay: 10s
    networks:
      - frontend
      - backend


volumes:
  postgres_data:

networks:
  frontend:
  backend:
